### PR TITLE
WT-8981 Enable evergreen testing for RHEL8 on PPC

### PR DESCRIPTION
**Pull request by mickgraham-mongodb**
WT-8981 Enable evergreen testing for RHEL8 on PPC. Remove build variant for Ubuntu on PPC. To work around the instances we use, this PR includes a retry for pread and disabling mmap.

### CODE REVIEW

**Review by keitharnoldsmith — COMMENTED** [review]
I've left a couple of minor suggestions about the code changes in `posix_file_read()`.

But more generally, I'm wondering if it would be worthwhile to always retry on a `pread()` failure, rather than making that special code for RHEL on PPC. In other places we wrap system calls in the `WT_SYSCALL_RETRY()` macro which will retry up to 10 times if underlying system call fails. 

One reason we don't use `WT_SYSCALL_RETRY()` with `pread()` is that the macro assumes that the system call returns zero on success, and `pread()` return a non-zero positive value on success. But I don't know if there were additional reasons we chose not to retry `pread()`. Maybe @agorrod remembers or knows something about this?

**Comment by mickgraham-mongodb** [comment]
> > The RHEL PPC platform does not have ZSTD. Strip it out.
> 
> Should we make sure we never enable zstd in a test/format configuration for this platform? Or is it taken care of already? Is the long-term solution to have zstd on the platform (BUILD ticket)?

With the way the evergreen.yaml file is setup, its easier to remove it from here as there are a lot of common variable setups that it shares. I've left it this way as to not increase the size of the yaml file.

File: src/os_posix/os_fs.c [hunk 1]
```diff
@@ -422,10 +422,25 @@ __posix_file_read(
     /* Break reads larger than 1GB into 1GB chunks. */
     for (addr = buf; len > 0; addr += nr, len -= (size_t)nr, offset += nr) {
         chunk = WT_MIN(len, WT_GIGABYTE);
+#ifdef RHEL_PPC
+        if ((nr = pread(pfh->fd, addr, chunk, offset)) <= 0) {
```
[the remark this answers is not in the document]
Comment (mickgraham-mongodb) [reply src/os_posix/os_fs.c:426]: I had it that way originally but it was just very difficult to read. I found it cleaner to have it explicit.

**Comment by etienneptl** [comment]
> With the way the evergreen.yaml file is setup, its easier to remove it from here as there are a lot of common variable setups that it shares. I've left it this way as to not increase the size of the yaml file.

Ok I get it and double-checked that in test/format we don't enable it, thanks!

File: src/os_posix/os_fs.c [hunk 2]
```diff
@@ -422,10 +422,25 @@ __posix_file_read(
...
+            /* Retry as the RHEL PPC variants we use have false failures that occur on pread. */
+            __wt_verbose(session, WT_VERB_READ,
+              "%s: handle-read: pread: failed to read %" WT_SIZET_FMT " bytes at offset %" PRIuMAX
+              ", will retry",
+              file_handle->name, chunk, (uintmax_t)offset);
+            if ((nr = pread(pfh->fd, addr, chunk, offset)) <= 0)
```
Comment (agorrod) [inline src/os_posix/os_fs.c:432]: This feels somewhat similar to the WT_SYSCALL_RETRY macro - could we reuse that instead?

**Review by agorrod — COMMENTED** [review]
I've left an annoying comment - I'll be happy to talk this through once you are back from vacation.

File: src/os_posix/os_fs.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (keitharnoldsmith) [reply src/os_posix/os_fs.c:426]: That's fine. I don't have any problem leaving the `#ifdef` the way you've got it.

**Comment by mickgraham-mongodb** [comment]
Updated the code to retry all the time (not just on RHEL PPC) using the WT_SYSCALL_RETRY macro based on Keith's comment and a discussion with Alex. We don't think there are critical performance paths that would be impacted by retries in pread.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM!

Clever way to use WT_SYSCALL_RETRY with a call that can return non-zero on success.

**Pull request merged**